spec/features/checkout_spec.rb

```diff
@@ -121,9 +121,13 @@
         fill_in "Gift Code", :with => "foobar"
         click_button "Save and Continue"
 
-        within '[data-hook="order-payment"]' do
-          expect(page).to have_content("Gift Card ($5.00)")
-        end
+        # Note: Even though page loads for order detail
```

Comment:
@pikender We can use wait_for_ajax. It can solve this problem.
Reference link provided below: https://github.com/spree/spree/blob/master/core/lib/spree/testing_support/capybara_ext.rb#L105